src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/HubProtocolExtensions.cs

```diff
@@ -1,19 +1,15 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.IO;
-
 namespace Microsoft.AspNetCore.SignalR.Internal.Protocol
 {
     public static class HubProtocolExtensions
     {
         public static byte[] WriteToArray(this IHubProtocol hubProtocol, HubMessage message)
         {
-            using (var ms = new LimitArrayPoolWriteStream())
-            {
-                hubProtocol.WriteMessage(message, ms);
-                return ms.ToArray();
-            }
+            var writer = new MemoryBufferWriter();
```

Comment:
MemoryBufferWriter needs some perf thought put into, along with unit tests if it is going to be used everywhere. I wrote it pretty quickly to get unit tests passing